hw14/election.py

```diff
@@ -0,0 +1,321 @@
+# Name: Sydney Decoto
+# CodeFellows Python F2: SEA-C42
+# Homework 14: Election prediction
+
+import csv
+import os
+import time
+
+
+def read_csv(path):
+    """
+    Reads the CSV file at path, and returns a list of rows. Each row is a
+    dictionary that maps a column name to a value in that column, as a string.
+    """
+    output = []
+    for row in csv.DictReader(open(path)):
+        output.append(row)
+    return output
+
+
+###############################################################################
+# Problem 1: State edges
+###############################################################################
+
+def row_to_edge(row):
+    """
+    Given an *ElectionDataRow* or *PollDataRow*, returns the
+    Democratic *Edge* in that *State*.
+    """
+    return float(row["Dem"]) - float(row["Rep"])
+
+
+def state_edges(election_result_rows):
+    """
+    Given a list of *ElectionDataRow*s, returns *StateEdge*s.
+    The input list has no duplicate *States*;
+    that is, each *State* is represented at most once in the input list.
+    """
+    d = {}
+    for row in election_result_rows:
+        state = row['State']
+        dem = row['Dem']
+        rep = row['Rep']
+
+        d[state] = float(dem) - float(rep)
+    return d
+
+###############################################################################
+# Problem 2: Find the most recent poll row
+###############################################################################
+
+
+def earlier_date(date1, date2):
+    """
+    Given two dates as strings (formatted like "Oct 06 2012"), returns True if
+    date1 is before date2.
+    """
+    return (time.strptime(date1, "%b %d %Y") < time.strptime(date2, "%b %d %Y"))
+
+
+def most_recent_poll_row(poll_rows, pollster, state):
+    """
+    Given a list of *PollDataRow*s, returns the most recent row with the
+    specified *Pollster* and *State*. If no such row exists, returns None.
+    """
+
+    returndata = 'None'
+    checkdate = 'Jan 01 1900'
+    for i in range(len(poll_rows)):
+        if state in poll_rows[i]['State'] and \
+           pollster in poll_rows[i]['Pollster']:
+            if earlier_date(checkdate, poll_rows[i]['Date']):
+                returndata = poll_rows[i]
+        else:
+            continue
+    if returndata == 'None':
```

Comment:
Again if you initialize returndata to None not 'None', you can just return returndata here without the if/else. Variables in python are dynamically typed, so there's no issue changes the type that's assigned to a variable.